controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -832,9 +838,13 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 
 	// Create the hostFirmwareSettings resource with same host name/namespace if it doesn't exist
 	if info.host.Name != "" {
-		if err = r.createHostFirmwareSettings(info); err != nil {
-			info.log.Info("failed creating hostfirmwaresettings")
-			return actionError{errors.Wrap(err, "failed creating hostFirmwareSettings")}
+		if info.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeDisabled && !info.host.DeletionTimestamp.IsZero() {
```

Comment:
I don't think we need to worry about the cleaning mode here. If the host is being deleted then there's no point creating a HostFirmwareSetting CR for it.

Ditto for HardwareData, so let's handle that at the same time.